staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -654,9 +659,9 @@ func (o *DebugOptions) generatePodCopyWithDebugContainer(pod *corev1.Pod) (*core
 			Name:                     name,
 			TerminationMessagePolicy: corev1.TerminationMessageReadFile,
 		}
-		defer func() {
+		appendDebugContainer = func() {
 			copied.Spec.Containers = append(copied.Spec.Containers, *c)
-		}()
+		}
```

Comment:
This looks like it changed his behavior

Reply:
Thank you for your response

Reply:
Yes, we need this to ensure the generated debug container is added the the container list prior to the `o.Applier.Apply` occurs on line 692 below or otherwise the generated debug container isn't available to `o.Applier`.

Reply:
Nice idea - done in 861a2a011bbff8c863a252e71c6c4e711d14b871